src/compute/src/server.rs

```diff
@@ -217,6 +223,41 @@ impl mz_cluster::types::AsRunnableWorker<ComputeCommand, ComputeResponse> for Co
     }
 }
 
+/// Set the current thread's core affinity, based on the given `worker_id`.
+fn set_core_affinity(worker_id: usize) {
+    let Some(mut core_ids) = core_affinity::get_core_ids() else {
+        warn!(worker_id, "unable to get core IDs for setting affinity");
+        return;
+    };
+
+    // The `get_core_ids` docs don't say anything about a guaranteed order of the returned Vec,
+    // so sort it just to be safe.
+    core_ids.sort_unstable_by_key(|i| i.id);
+
+    // On multi-process replicas `worker_id` might be greater than the number of available cores.
+    // However, we assume that we always have at least as many cores as there are local workers.
+    // Violating this assumption is safe but might lead to degraded performance due to skew in core
```

Comment:
Can we log an error here if this assumption is violated?

Reply:
Not easily because we don't know how many local workers there are! Getting this information wired through is possible but I couldn't see a clean way to do so. Maybe I'm missing something.

I think it's fine to assume that our production instance sizes are vetted by compute people and we would notice ones that have more workers than cores available.

Reply:
That’s fine with me